src/libc/src/eacsl.c

```diff
@@ -0,0 +1,198 @@
+  va_list args;
+  va_start(args, count);
+
+  int result = 0;
+  for (size_t i = 0; i < 2 && i < count; ++i) {
+    result += __e_acsl_contract_get_behavior_assumes(c, va_arg(args, int));
+  }
+
+  va_end(args);
+  return result;
+}
+
+int __e_acsl_contract_partial_count_all_behaviors(const __e_acsl_contract_t *c) {
+  int result = 0;
+  for (int i = 0; i < c->char_count && result < 2; ++i) {
+    // Counting bits set with Kernighan's algorithm, but stopping at two
+    // bits set.
+    // cf. <https://graphics.stanford.edu/~seander/bithacks.html#CountBitsSetKernighan>
+    unsigned char assumes_cell = c->assumes[i];
+    for (; assumes_cell && result < 2; ++result) {
+      assumes_cell &= assumes_cell - 1;
+    }
+  }
+  return result;
+}
+
+/* Assertions */
+
+/* Assert */
+void __e_acsl_assert(int predicate, __e_acsl_assert_data_t *data) {
+    owi_assert(predicate);
+}
+
+/* Assert data registration */
+void __e_acsl_assert_register_bool(__e_acsl_assert_data_t *data, const char *name,
+                                int is_enum, _Bool value) {}
+void __e_acsl_assert_register_char(__e_acsl_assert_data_t *data, const char *name,
+                                int is_enum, char value) {}
+void __e_acsl_assert_register_schar(__e_acsl_assert_data_t *data, const char *name,
+                                 int is_enum, signed char value) {}
+void __e_acsl_assert_register_uchar(__e_acsl_assert_data_t *data, const char *name,
+                                 int is_enum, unsigned char value) {}
+void __e_acsl_assert_register_int(__e_acsl_assert_data_t *data, const char *name,
+                               int is_enum, int value) {}
+void __e_acsl_assert_register_uint(__e_acsl_assert_data_t *data, const char *name,
+                                int is_enum, unsigned int value) {}
+void __e_acsl_assert_register_short(__e_acsl_assert_data_t *data, const char *name,
+                                 int is_enum, short value) {}
+void __e_acsl_assert_register_ushort(__e_acsl_assert_data_t *data, const char *name,
+                                  int is_enum, unsigned short value) {}
+void __e_acsl_assert_register_long(__e_acsl_assert_data_t *data, const char *name,
+                                int is_enum, long value) {}
+void __e_acsl_assert_register_ulong(__e_acsl_assert_data_t *data, const char *name,
+                                 int is_enum, unsigned long value) {}
+void __e_acsl_assert_register_longlong(__e_acsl_assert_data_t *data, const char *name,
+                                    int is_enum, long long value) {}
+void __e_acsl_assert_register_ulonglong(__e_acsl_assert_data_t *data,
+                                     const char *name, int is_enum,
+                                     unsigned long long value) {}
+// void __e_acsl_assert_register_mpz(__e_acsl_assert_data_t *data, const char *name,
+//                                int is_enum, const __e_acsl_mpz_t value) {}
```

Comment:
If those are unused, can we remove the comments ?

Reply:
We have to comment them out because we cannot use GMP in webassembly. These functions can be generated by E-ACSL, though.

Reply:
Could you remove the comments and add an example where E-ACSL generates such functions ?